src/col_tests/utils/inst_dissolution.py

```diff
@@ -0,0 +1,87 @@
+import pandas as pd
+from numpy import trapz
+from scipy.integrate import simpson
+
+from src.col_tests.utils.test_settings import get_test_settings
+
+
+def integrate(y, x, method):
+    """ Approximates integral for inst. dissolution equation.
+
+    Args
+        y:          Input array over which to integrate
+        x:          Array of sample points corresponding to the input array
+        method:     Str. Approximation rule. Either 'trapezoidal' or 'simpson'
+
+    Returns
+        res:        Float. Approximated integral value
+    """
+    if method == "trapezoidal":
+        res = trapz(y, x)
+    else:
+        res = simpson(y, x)
+
+    return res
+
+
+def calculate_inst_dissolution(col_groups, element, test_input, test_type, method):
+    """ Calculates instantaneous dissolution for test columns.
+
+    Args
+        col_groups:     DataFrameGroupBy. DataFrame of input data grouped by test columns
+        element:        Str. Chemical element for which to calculate. Either 'Ca' or 'Mg'
+        test_input:     Float. Test parameter value
+        test_type:      Str. Type of the test. Either 'instantaneous' or 'overdosing'
+        method:         Str. Approximation rule. Either 'trapezoidal' or 'simpson'
+
+    Returns
+        inst_dist_list: List of instantaneous dissolution values for each test column
+    """
+    inst_diss_list = []
+    param_settings = get_test_settings(test_type)
+
+    for col, col_df in col_groups:
+        col_df.sort_values("Depth_m", inplace=True)
+        ys = col_df[element + "_mg/l"].values
+        xs = col_df["Depth_m"].values
+        xmin, xmax = xs.min(), xs.max()
+
+        res = integrate(ys, xs, method)
+        inst_diss_pct = param_settings['percent_factor'] * res / (param_settings['od_factor'][col] * test_input * (xmax - xmin))
```

Comment:
I think this calculation is more clearly expressed as

$$ID_{pct} = \frac{100 * res}{D F_x (xmax - xmin)}$$

Where $D$ is the dose of lime and $F_x$ is the fraction of Ca or Mg by mass (see steps 1 to 4 in the main review text).

Sorry for the confusion here - I realise this is not obvious in my code either (it's only while writing the code that I figured out the tests were basically the same).